src/utils/solidity/getContractWithPath.ts

```diff
@@ -0,0 +1,24 @@
+import * as R from 'ramda';
+import * as Eth from 'web3-eth';
+import { Address } from '~/utils/types';
+import { getGlobalEnvironment, Environment } from '~/utils/environment';
+
+export type GetContractWithPathFunction = (
+  relativePath: string,
+  address: Address,
+  environment?: Environment,
+) => typeof Eth.Contract;
+
+export const getContractWithPath: GetContractWithPathFunction = R.memoizeWith(
```

Comment:
Whats the difference to `getContract`? The possibility to not have to use `Contracts` enum? Why not just add these contracts to the enum then?

Reply:
Yeah I think I was just trying to avoid polluting the enum with all the testing contracts but there's no harm in doing that really. We can factor out getContractWithPath if we want then